### PR TITLE
Better memory management

### PR DESCRIPTION
**Pull request by Arkaeriit**
This PR fixes a memory error and use makes the manipulation of structs more idiomatic.

### CODE REVIEW

**Comment by Arkaeriit** [comment]
The bug fixed in [1ee2653](https://github.com/THEGOLDENPRO/aghpb.c/pull/2/commits/1ee26539d97d00892de9c9e30a56001e2f9da69d) was a pretty bad one that could have caused a lot of damage if this API was used in a bigger project. I noticed it thanks to valgrind, but I noticed afterward that I could have caught it with [cppcheck](http://cppcheck.net/), a very good static code analysis for C and C++.
To summarize it, you allocated some memory on the stack with stuff like `char book_name[1000];` and then used `book_name` in the return value. This is an error as `book_name` is allocated on the stack and thus only valid inside of `aghpb_random`.

File: aghpb.c
Comment (THEGOLDENPRO) [inline aghpb.c:1]: amazing work ✨

**Comment by THEGOLDENPRO** [comment]
I see, thanks so much for the detailed explaination 👍 that helps a lot.
